vtex-apps/order-manager
# Update local orderForm when ids differ from server

#### What problem is this solving?
After you finish your order, the local orderForm won't be updated to use a "new" one, since the we were giving priority to using the local one instead of the OF sent from the API, resulting in the user having the same cart as they had before they finished the order even though it shouldn't.

To fix this, we will now replace the local orderForm with the value from the server if the id from the local OF is different from the one sent from the server.

[Related Clubhouse story](https://app.clubhouse.io/vtex/story/32316/atualizar-orderform-offline-quando-api-retornar-um-novo).

#### How should this be manually tested?
[Workspace](https://reneworderform--checkoutio.myvtex.com/).

- Add some items to your cart.
- Proceed to checkout and finish your order.
- On the order placed screen, your cart should change from the number of items added to zero.

#### Checklist/Reminders


#### Type of changes


✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements
